prow/scripts/cluster-integration/kyma-gke-end-to-end-test.sh

```diff
@@ -0,0 +1,302 @@
+    COMMON_NAME=$(echo "gkeint-rel-${RANDOM_NAME_SUFFIX}" | tr "[:upper:]" "[:lower:]")
+else
+    # Otherwise (master), operate on triggering commit id
+    readonly COMMIT_ID=$(cd "$KYMA_SOURCES_DIR" && git rev-parse --short HEAD)
+    COMMON_NAME=$(echo "gkeint-commit-${COMMIT_ID}-${RANDOM_NAME_SUFFIX}" | tr "[:upper:]" "[:lower:]")
+    KYMA_INSTALLER_IMAGE="${DOCKER_PUSH_REPOSITORY}${DOCKER_PUSH_DIRECTORY}/gke-integration/${REPO_OWNER}/${REPO_NAME}:COMMIT-${COMMIT_ID}"
+    export KYMA_INSTALLER_IMAGE
+fi
+
+
+### Cluster name must be less than 40 characters!
+export CLUSTER_NAME="${COMMON_NAME}"
+
+### For provision-gke-cluster.sh
+export GCLOUD_PROJECT_NAME="${CLOUDSDK_CORE_PROJECT}"
+export GCLOUD_COMPUTE_ZONE="${CLOUDSDK_COMPUTE_ZONE}"
+
+#Local variables
+DNS_SUBDOMAIN="${COMMON_NAME}"
+KYMA_SCRIPTS_DIR="${KYMA_SOURCES_DIR}/installation/scripts"
+KYMA_RESOURCES_DIR="${KYMA_SOURCES_DIR}/installation/resources"
+
+INSTALLER_YAML="${KYMA_RESOURCES_DIR}/installer.yaml"
+INSTALLER_CONFIG="${KYMA_RESOURCES_DIR}/installer-config-cluster.yaml.tpl"
+INSTALLER_CR="${KYMA_RESOURCES_DIR}/installer-cr-cluster.yaml.tpl"
+
+#Used to detect errors for logging purposes
+ERROR_LOGGING_GUARD="true"
+
+shout "Authenticate"
+date
+init
+DNS_DOMAIN="$(gcloud dns managed-zones describe "${CLOUDSDK_DNS_ZONE_NAME}" --format="value(dnsName)")"
+
+if [[ "$BUILD_TYPE" != "release" ]]; then
+    shout "Build Kyma-Installer Docker image"
+    date
+    CLEANUP_DOCKER_IMAGE="true"
+    "${TEST_INFRA_CLUSTER_INTEGRATION_SCRIPTS}/create-image.sh"
+fi
+
+shout "Reserve IP Address for Ingressgateway"
+date
+GATEWAY_IP_ADDRESS_NAME="${COMMON_NAME}"
+GATEWAY_IP_ADDRESS=$(IP_ADDRESS_NAME=${GATEWAY_IP_ADDRESS_NAME} "${TEST_INFRA_CLUSTER_INTEGRATION_SCRIPTS}/reserve-ip-address.sh")
+CLEANUP_GATEWAY_IP_ADDRESS="true"
+echo "Created IP Address for Ingressgateway: ${GATEWAY_IP_ADDRESS}"
+
+
+shout "Create DNS Record for Ingressgateway IP"
+date
+GATEWAY_DNS_FULL_NAME="*.${DNS_SUBDOMAIN}.${DNS_DOMAIN}"
+CLEANUP_GATEWAY_DNS_RECORD="true"
+IP_ADDRESS=${GATEWAY_IP_ADDRESS} DNS_FULL_NAME=${GATEWAY_DNS_FULL_NAME} "${TEST_INFRA_CLUSTER_INTEGRATION_SCRIPTS}/create-dns-record.sh"
+
+
+shout "Reserve IP Address for Remote Environments"
+date
+REMOTEENVS_IP_ADDRESS_NAME="remoteenvs-${COMMON_NAME}"
+REMOTEENVS_IP_ADDRESS=$(IP_ADDRESS_NAME=${REMOTEENVS_IP_ADDRESS_NAME} "${TEST_INFRA_CLUSTER_INTEGRATION_SCRIPTS}/reserve-ip-address.sh")
+CLEANUP_REMOTEENVS_IP_ADDRESS="true"
+echo "Created IP Address for Remote Environments: ${REMOTEENVS_IP_ADDRESS}"
+
+
+shout "Create DNS Record for Remote Environments IP"
+date
+REMOTEENVS_DNS_FULL_NAME="gateway.${DNS_SUBDOMAIN}.${DNS_DOMAIN}"
+CLEANUP_REMOTEENVS_DNS_RECORD="true"
+IP_ADDRESS=${REMOTEENVS_IP_ADDRESS} DNS_FULL_NAME=${REMOTEENVS_DNS_FULL_NAME} "${TEST_INFRA_CLUSTER_INTEGRATION_SCRIPTS}/create-dns-record.sh"
+
+
+shout "Provision cluster: \"${CLUSTER_NAME}\""
```

Comment:
If this will be exexuted as a periodic job, you should at the beginning remove previous clusters and all resources.

Reply:
I will add a call to the cleanup function before the creation of the cluster
```
shout "Cleanup"
date
cleanup
```